src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -437,51 +468,120 @@ public void delete(String modelId, ActionListener<DeleteModelResponse> listener)
                 return;
             }
 
+            StepListener<GetModelResponse> getModelStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> blockModelIdStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> clearModelMetadataStep = new StepListener<>();
+            StepListener<DeleteResponse> deleteModelFromIndexStep = new StepListener<>();
+            StepListener<RemoveModelFromCacheResponse> clearModelFromCacheStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> unblockModelIdStep = new StepListener<>();
+
+            // Get Model to check if model is in TRAINING
+            get(modelId, ActionListener.wrap(getModelStep::onResponse, exception -> {
+                if (exception instanceof ResourceNotFoundException) {
+                    listener.onResponse(new DeleteModelResponse(modelId, "failed", exception.getMessage()));
```

Comment:
Instead of passing the resource not found exception up the stack, can we just say "Unable to delete the model: model does not exist"?